src/findNodes.ts

```diff
@@ -16,112 +18,145 @@ function findChildrenWithIDs(children: Array<SGNode>): Map<string, Range> {
     return foundIDs;
 }
 
+function getFilteredScopes(program: Program) {
```

Comment:
This exclusion logic is not correct. Each pattern operates based on the results of the previous pattern. For example:

```jsonc
[
    //exclude all vendor scripts
    "source/vendor/**/*",
    //except don't exclude rodash
    "!source/vendor/rodash/**/*",
    //except DO exclude this specific rodash file
    "source/vendor/rodash/_all.bs"
]
```